### DISCUSSION TITLE
Two questions on path.glob-tree behavior

### DISCUSSION THREAD

**Discussion by gabrieldevillers** [bfgroup/b2#432] [General]
I have encountered the following issue: https://stackoverflow.com/questions/79284424/impossible-to-link-against-target-defined-in-another-jamfile-relative-sources-m

### Environment and version details

* Operating System+version: **Windows 11**
* Compiler+version: **MSVC 19.40.33813**
* Shell: **Git Bash**
* B2 Version: `B2 Version 4.10. OS=NT.` ; `B2 4.10-git`
* B2 Configuration: will provide it if needed

### Brief problem description

1) rule `path.glob-tree` seems to return paths that are relative to working directory, not the jamfile containing it. Is this intended ? If yes this should be documented I think.
2) the workaround I describe in the SO thread using `path-constant` causes `path.glob-tree` to return rooted paths, so I had to write the rule `my-relative-path` to get relative paths again. Is this normal (I guess it is because path-constant return an absolute path) ?
2.1) Is there a built-in alternative to `my-relative-path` ?

**Comment by grisumbras** [comment]
1. `glob-tree` is a project rule, so it has access to the current project. `path.glob-tree` doesn't have access to the current project, because it's not a project rule, and in fact it doesn't even know if there is a current project. So, the behaviour is kind of obvious if you think about it, but I agree that it's better to document such things.
2. Project rule `rule glob-tree-ex ( paths + : wildcards + : excludes * )` is what you are looking for. It's not documented, and it definitely should be.

**Reply by gabrieldevillers** [reply]
Thank you very much, `glob-tree-ex` seems indeed to be what I need.
